github-review.el

```diff
@@ -476,29 +427,37 @@ ACC is an alist accumulating parsing state."
                                    (github-review-a-assoc 'sha (match-string 3 url)))))
            commit-alist))))
 
-(defun github-review-save-diff (pr-alist diff)
-  "Save a DIFF (string) to a temp file named after pr specified by PR-ALIST."
-  (find-file (format "%s/%s___%s___%s.diff"
-                     github-review-review-folder
-                     (github-review-a-get pr-alist 'owner)
-                     (github-review-a-get pr-alist 'repo)
-                     (github-review-a-get pr-alist 'num)))
-  (erase-buffer)
-  (insert diff)
-  (save-buffer)
-  (github-review-mode))
-
-(defun github-review-save-commit-diff (commit-alist diff)
-  "Save a DIFF (string) to a temp file named after commit specified by COMMIT-ALIST."
-  (find-file (format "%s/%s___%s___commit___%s.diff"
-                     github-review-review-folder
-                     (github-review-a-get commit-alist 'owner)
-                     (github-review-a-get commit-alist 'repo)
-                     (github-review-a-get commit-alist 'sha)))
-  (erase-buffer)
-  (insert diff)
-  (save-buffer))
-
+(defun github-review-save-diff (id-alist diff ids)
+  "Save a DIFF (string) to a temp file named after pr or commit sha specified by ID-ALIST.
+IDS stands for alist of the diff and its parent's sha:s, when supplied the two becomes
+buffer-local variables"
+  (let* ((is-pr (github-review-is-pr id-alist))
+	 (buffer-read-only))
+    (find-file (format (if is-pr "%s/%s___%s___%s.diff" "%s/%s___%s___commit___%s.diff")
+                       github-review-review-folder
+                       (github-review-a-get id-alist 'owner)
+                       (github-review-a-get id-alist 'repo)
+                       (github-review-a-get id-alist (if is-pr 'num 'sha))))
+    (setq buffer-read-only nil)
+    (erase-buffer)
+    (insert diff)
+    (when github-review-save-commit-id-in-buffer
+      (when (not (null ids))
+	(set-variable 'gh-commit-id       (first  ids) t)
+	(set-variable 'gh-first-parent-id (second ids) t)
+	(insert
```

Comment:
What is the rationale for setting them here instead of making them buffer local? Is it because you want this to persist across session? The downside is that the user will be typically prompted to accept those on a per buffer basis which increases friction to start any review

Reply:
@andrelkin

Reply:
> Is it because you want this to persist across session?
Right.

The prompting downside is just one time as the var:s can be added to
`safe-local-variable-values` at the first file opening.

Reply:
To follow up with the user-customizable file extension, it more makes sense
with a hook at the end of `(github-review-save-diff)`. That is my preference block
```
+    (when github-review-save-commit-id-in-buffer
+      (when (not (null ids))
+	(set-variable 'gh-commit-id       (first  ids) t)
+	(set-variable 'gh-first-parent-id (second ids) t)
``` would go into my personal hook.

Reply:
Deu?

Reply:
Testing sorry